scripts/linux/electron-installer-appimage.sh

```diff
@@ -102,17 +101,14 @@ else
   exit 1
 fi
 
-if [ -z "$TMPDIR" ]; then
-  TMPDIR=$(mktemp -d)
-fi
-mkdir -p "$TMPDIR"
+APPIMAGE_TMPDIR=$(mktemp --dry-run -d -t "$ARGV_APPLICATION_NAME-linux-$ARGV_ARCHITECTURE-XXXX")
```

Comment:
So as far as I can see (their Makefiles are huge, and didn't go through all of it), they have an `INSTALL` utility that gets called by scripts as they need, and the set the output location to the `dl` directory. I like that approach and I'm happy to go with it!

I'll create a generic script that performs downloads (and takes a MD5 argument to verify the files), tries to cache if necessary, and will make all scripts call it.